pytorch_lightning/callbacks/base.py

```diff
@@ -16,8 +16,16 @@
 Abstract base class used to build new callbacks.
 
 """
-
 import abc
+from typing import TYPE_CHECKING, Any, Dict, List, Union
+
+from torch.optim.optimizer import Optimizer
+
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.core.optimizer import LightningOptimizer
+
+if TYPE_CHECKING:
```

Comment:
For context on this, see https://github.com/PyTorchLightning/pytorch-lightning/issues/5023#issuecomment-743745278